### PR TITLE
Set SKIP_INSTALL=YES for all generated targets

### LINKED ISSUE #3007: 0.36.0.beta.1 breaks archiving project as Mac App Archive
#### Summary
After installation of 0.36.0.1.beta I noticed being unable to archive a Mac App project as a Mac application. Xcode just offers me "Generic Xcode Archive".

#### Steps to Reproduce
* create an empty "Cocoa Application" project
* do nothing but archiving of the product. As you can see, Xcode offers a valid *Mac App Archive*:

valid Archive Type

* close this project and create a Podfile within the project folder using this example (you can choose any pod):

```
platform :osx, "10.9"
xcodeproj "Test"

pod 'CCNStatusItemView'
```
* now fire `pod install`
* open the workspace
* archive again

#### Expected Results
The same valid *Mac App Archive* as shown above.

#### Actual Results
I can just archive as *Generic Xcode Archive*:

genericxcodearchive

### PR DESCRIPTION
**Pull request by mrackwitz**
Fixes #3007.

### CODE REVIEW

**Comment by segiddins** [comment]
Is there a chance this will have unintended side effects when using Pods in a framework, or something?

**Comment by mrackwitz** [comment]
This could influence every build script, which rely on the build paths of generated build products.
If [`SKIP_INSTALL=YES`](https://developer.apple.com/library/ios/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html#//apple_ref/doc/uid/TP40003931-CH3-SW36) and [`DEPLOYMENT_LOCATION=YES`](https://developer.apple.com/library/ios/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html#//apple_ref/doc/uid/TP40003931-CH3-SW37), which is implied by [`ACTION=install`](https://developer.apple.com/library/ios/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html#//apple_ref/doc/uid/TP40003931-CH3-SW45), which is the verb given on the CLI to xcodebuild, then the build product is placed into `$TARGET_TEMP_DIR/UninstalledProducts`.
Usually, this happens for a framework or library target only when the target is build as an implicit target dependency of an app target, which the user archives from Xcode.
This [does **not** apply](https://github.com/CocoaPods/cocoapods-packager/blob/master/lib/cocoapods-packager/builder.rb#L172) for [cocoapods-packager](/CocoaPods/cocoapods-packager).

**Comment by segiddins** [comment]
@mrackwitz I'm not sure you've answered my question, which is, will this break things when you use CocoaPods with a framework target?

**Comment by mrackwitz** [comment]
It could because the AggregateXCConfig is shared with the user target.
But it's only a matter if the user actually want to install the framework to the system's framework paths (`$(LOCAL_LIBRARY_DIR)/Frameworks`) or use `INSTALL_PATH` in a different way, which applies probably only in rare cases.
We could avoid that by inlining the build setting to Pods.xcodeproj.

**Comment by segiddins** [comment]
So are you saying I shouldn't merge this PR now?

File: CHANGELOG.md [hunk 1]
```diff
@@ -67,6 +67,11 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Set `SKIP_INSTALL=YES` for all generated targets to avoid producing
+  *Generic Xcode Archives* on Archive.
```
Comment (segiddins) [inline CHANGELOG.md]: two spaces

**Comment by segiddins** [comment]
@kylef I'll merge if this gets your :+1:

**Pull request merged**